apps/tutorial_servo_control_can/TutorialServo.cpp

```diff
@@ -0,0 +1,28 @@
+#include "TutorialServo.h"
+#include "mbed.h"
+
+TutorialServo::TutorialServo(PinName servoPin) 
+	:_servoPwmOut(servoPin), _minPulseWidthInMs(1), _maxPulseWidthInMs(2), _servoRangeInDegrees(180)
+{
+	_minPulseWidthInMs = _minPulseWidthInMs / 1000;
+	_maxPulseWidthInMs = _maxPulseWidthInMs / 1000;
+	//default period of servo is 50Hz or 20 milliseconds 
+	_servoPwmOut.period(0.020);
+}
+
+void TutorialServo::setAngleRangeInDegrees(float degrees) {
+	_servoRangeInDegrees = degrees; 
+}
+
+void TutorialServo::setPulseWidthRangeInMs(float minPulseWidthInMs, float maxPulseWidthInMs) {
+	_minPulseWidthInMs = minPulseWidthInMs/1000;
+	_maxPulseWidthInMs = maxPulseWidthInMs/1000;
+}
+void TutorialServo::setServoPositionInDegrees(float degrees) {
+	if (degrees > 180) {

```

Comment:
the max shouldnt be har-coded as 180. what if the servo has a range of 270 degrees and i would like it to go to 270? This piece of code would make it go to position 180.

Reply:
I though servo ranges are a max of 180 ? Anyways I moved the constraint